src/Response.php

```diff
@@ -159,9 +156,8 @@ public function getReasonPhrase()
      */
     public function withStatus($code, $reasonPhrase = '')
     {
-        $this->validateStatus($code);
```

Comment:
Revert all changes after this line. There is a clear intention of not provide any kind of setter.